frontend/src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import { Routes, Route } from "react-router-dom";
+import RequestsPage from "./pages/requests/RequestsPage";
+import SmileNavBar from "./shared/components/SmileNavBar";
+import PatientsPage from "./pages/patients/PatientsPage";
+import SamplesPage from "./pages/samples/SamplesPage";
+import { useState } from "react";
+
+function App() {
+  const [userEmail, setUserEmail] = useState<string | null>(null);
+
+  return (
+    <main id="main" className="main">
+      <SmileNavBar userEmail={userEmail} />
+      <Routes>
+        <>
+          <Route path="/" element={<RequestsPage />}>
+            <Route path=":igoRequestId" />
+          </Route>
+          <Route path="/requests/" element={<RequestsPage />}>
+            <Route path=":igoRequestId" />
+          </Route>
+          <Route
+            path="/patients/"
+            element={<PatientsPage setUserEmail={setUserEmail} />}
+          >
+            <Route path=":cmoPatientId" />
+          </Route>
+          <Route path="/samples" element={<SamplesPage />} />
+        </>
+      </Routes>
+    </main>
+  );
+}
+
+export default App;
```

Comment:
Most of this code and routes were directly moved out of `frontend/src/index.tsx`. I moved them into App, a functional component, to use hooks. 

Purpose of the `userEmail` state: After user logs in for the first time, the login popup window would post a message to the opener window that includes the user's email. That way, the opener window's navbar can immediately display "Logged in as {userEmail}" and a logout button. Work to persist this will come in a later ticket.